CHANGELOG.md

```diff
@@ -3,6 +3,260 @@
+- [DataGrid] Fix `null` reference error in `GridVirtualScrollbar` (#15009) @MBilalShafi
+- [DataGrid] Fix failing CI (#15273) @MBilalShafi
+- [DataGrid] Fix filtering with `boolean` column type (#15252) @k-rajat19
+- [DataGrid] Fix overlay height in `autoHeight` mode (#15201) @cherniavskii
+- [DataGrid] Fix resizing right pinned column (#15107) @KenanYusuf
+- [DataGrid] Fix right column group header border (#15105) @KenanYusuf
+- [DataGrid] Fix scroll jump when holding down arrow keys (#15164) @arminmeh
+- [DataGrid] Focus next row when the focused row is deleted (#15061) @cherniavskii
+- [DataGrid] Improve row selection propagation trigger (#15184) @MBilalShafi
+- [DataGrid] Move `rowGroupingModelChange` handler to respective hook (#15124) @MBilalShafi
+- [DataGrid] Preprocess edit cell props on backspace/delete (#15151) @KenanYusuf
+- [DataGrid] Prevent error when deleting the last row (#15146) @dpak-maurya
+- [DataGrid] Set default overlay height in flex parent layout (#15202) @cherniavskii
+- [DataGrid] refactor: baseMenuList & baseMenuItem (#15049) @romgrk
+- [DataGrid] refactor: remove more material imports (#15063) @romgrk
+- [DataGrid] - Fix grid overlay aligment with scroll for rtl (#15072) @kalyan90
+- [I10n] Improve Polish (pl-PL) locale (#15227) @belkocik
+
+#### `@mui/x-data-grid-pro@8.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@8.0.0-alpha.0`, plus:
+
+- [DataGridPro] Add list view tests (#15043) @KenanYusuf
+- [DataGridPro] Fix column pinning layout (#14966) @cherniavskii
+- [DataGridPro] Apply default properties if they are not passed in a reorder column (#14473) @k-rajat19
+- [DataGridPro] Toggle row expansion with `Enter` key in "tree data" (#15271) @k-rajat19
+
+#### `@mui/x-data-grid-premium@8.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@8.0.0-alpha.0`, plus:
+
+- [DataGridPremium] Fix incorrect rows selection count when selection propagation is enabled with row grouping (#15216) @arminmeh
+- [DataGridPremium] Keep focus on the grouping cell on space bar press (#15108) @k-rajat19
+- [DataGridPremium] Server-side data source with row grouping (#13826) @MBilalShafi
+
+### Date and Time Pickers
+
+#### Breaking changes
+
+- The default DOM structure of the field has changed.
+  Before version `v8.x`, the fields' DOM structure consisted of an `<input />`, which held the whole value for the component, but unfortunately presents a few limitations in terms of accessibility when managing multiple section values.
+  Starting with version `v8.x`, all the field and picker components come with a new DOM structure that allows the field component to set aria attributes on individual sections, providing a far better experience with screen readers.
+  **Fallback to the non-accessible DOM structure:**
```

Comment:
This section is empty without the code snippet :laughing: 

For the new DOM structure you can just remove everything under `**Fallback to the non-accessible DOM structure:**`.

And I would add a link to the right section of the migration guide for each breaking change that was shortened.
Something like:

See [Migration guide—New DOM structure for the field](https://next.mui.com/x/migration/migration-pickers-v7/#new-dom-structure-for-the-field) for more details.

Reply:
Added the migration guide to the changes: https://github.com/mui/mui-x/blob/186313072fd5714ccbadb95042f6a8ed7d5871a0/CHANGELOG.md

Had to manually add them though, and slightly reworked the text in some cases. 

This is not the latest version from master, I'll do another pass when we have a set release date

Reply:
> Had to manually add them though, and slightly reworked the text in some cases.

I'll add them on by PR descriptions for the next BCs :+1: